api/api_controller.go

```diff
@@ -1098,6 +1106,9 @@ func (c *Controller) ObjectsUploadObjectHandler() objects.UploadObjectHandler {
 					StorageNamespace: repo.StorageNamespace,
 				},
 			})
+		if errors.Is(err, db.ErrNotFound) {
```

Comment:
This needs a test.

Reply:
talking about a case where a branch exists at the start of the operation and deleted at the end?

Reply:
A test for this error being returned.  It didn't use to be returned and all tests passed, so now we could use a test that goes through this branch.